File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.cpp
#include <string>
#include <vector>
#include <set>
#include <iostream>
using namespace std;

int solution(vector<int> elements) {
    int answer = 0;
    set<int> record;
    
    for(int startIdx = 0; startIdx < elements.size(); ++startIdx)
    {
        int sum = 0;
        int idx = startIdx;
        int length = 0;
        while(length < elements.size())
        {
            sum += elements[idx % elements.size()];
            
            record.insert(sum);
            ++length;
            ++idx;
        }
    }
    
    answer = record.size();
    return answer;
}

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.java
import java.util.*;

class Solution {
    public int solution(int[] elements) {
        Set<Integer> set = new HashSet<>();
        for (int len = 1; len <= elements.length; len++) {
            for (int i = 0; i < elements.length; i++) {
                int sum = 0;
                for (int j = i; j < i + len; j++) {
                    sum += elements[j % elements.length];
                }
                set.add(sum);
            }
        }
        return set.size();
    }
}